Sources/Ink/Internal/Link.swift

```diff
@@ -19,20 +20,35 @@ internal struct Link: Fragment {
 
         if reader.currentCharacter == "(" {
             reader.advanceIndex()
-            let url = try reader.read(until: ")")
-            return Link(target: .url(url), text: text)
+            let url = try reader.readCharacters(matching: \.isLegalInURL)
```

Comment:
```suggestion
            let url = try? reader.readCharacters(matching: \.isLegalInURL)
```

This helps fix the new failure of CommonMark tests 483 and 563:
"Both the title and the destination may be omitted" ([spec](https://spec.commonmark.org/0.29/#example-483))
"Inline links also take precedence" ([spec](https://spec.commonmark.org/0.29/#example-563))

If both title and destination are missing, then this line encounters ")" immediately and throws, which ends link parsing. So instead of throwing, we assign `nil` to the url if it is not present, and use a default value of `""` on line 33.